### PR TITLE
Add debug logs for snapshotting task file properties

### PR DESCRIPTION
**Pull request by lptr**

### CODE REVIEW

File: subprojects/core/src/main/java/org/gradle/api/internal/changedetection/state/CacheBackedTaskHistoryRepository.java [hunk 1]
```diff
@@ -208,6 +208,10 @@ public FileCollectionSnapshot transformEntry(String propertyName, FileCollection
 
     private static FileCollectionSnapshot snapshotDiscoveredInputs(Task task, InputNormalizationStrategy normalizationStrategy, Collection<File> discoveredInputs, FileCollectionSnapshotterRegistry snapshotterRegistry, FileCollectionFactory fileCollectionFactory) {
         GenericFileCollectionSnapshotter snapshotter = snapshotterRegistry.getSnapshotter(GenericFileCollectionSnapshotter.class);
+        LOGGER.debug("Snapshotting discovered inputs for {}", task);
```
Comment (wolfs) [inline subprojects/core/src/main/java/org/gradle/api/internal/changedetection/state/CacheBackedTaskHistoryRepository.java]: Don't we want to log after we are sure we have discovered inputs?

File: subprojects/core/src/main/java/org/gradle/api/internal/changedetection/state/CacheBackedTaskHistoryRepository.java [hunk 2]
```diff
@@ -316,6 +320,7 @@ private static boolean isOutputEntry(String path, NormalizedFileSnapshot fileSna
             FileCollectionSnapshot result;
             try {
                 FileCollectionSnapshotter snapshotter = snapshotterRegistry.getSnapshotter(propertySpec.getSnapshotter());
+                LOGGER.debug("Snapshotting property '{}' for {}", propertySpec, task);
```
Comment (wolfs) [inline subprojects/core/src/main/java/org/gradle/api/internal/changedetection/state/CacheBackedTaskHistoryRepository.java]: Should we really add `'` around the property spec? Currently the output looks like this:
```
Snapshotting property 'rootSpec$1 (RELATIVE)' for task ':jar'
```

File: subprojects/core/src/main/java/org/gradle/api/internal/changedetection/state/CacheBackedTaskHistoryRepository.java (hunk 1 above)
Comment (lptr) [inline subprojects/core/src/main/java/org/gradle/api/internal/changedetection/state/CacheBackedTaskHistoryRepository.java]: I've added cleaner logging.

File: subprojects/core/src/main/java/org/gradle/api/internal/changedetection/state/CacheBackedTaskHistoryRepository.java (hunk 2 above)
Comment (lptr) [inline subprojects/core/src/main/java/org/gradle/api/internal/changedetection/state/CacheBackedTaskHistoryRepository.java]: Removed quotes.

**Pull request merged**